File: 485. Max Consecutive Ones.java
// 485. Max Consecutive Ones
// TIme: O(N)
// Space: O(1)

// iterate and keep track of max value
public class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int cnt = 0, max = 0;
        for (int num : nums) {
            if (num == 1) {
                cnt += 1;
            } else {
                cnt = 0;
            }
            max = Math.max(max, cnt);
        }
        return max;
    }
}

File: 485. Max Consecutive Ones.py
# 485. Max Consecutive Ones
# Time: O(N)
# Space: O(1)

# use a cnt to accumulate 1s and a max_cnt to record the max length
class Solution:
    def findMaxConsecutiveOnes(self, nums: 'List[int]') -> 'int':
        max_cnt = 0
        cnt = 0
        for num in nums:
            if num == 1:
                cnt += 1
                if cnt > max_cnt:
                    max_cnt = cnt
            else:
                cnt = 0
        return max_cnt
